### ISSUE TITLE
Use Legacy Swift Language Version

### ISSUE THREAD

**Issue by loretoparisi** [eBay/NMessenger#51]
With Xcode 8.0 and Swift 3.0 using `Use Legacy Swift Language Version=YES` causes compiler to fail, even if this is needed since it Xcode tried to convert to modern Swift syntax.

**Comment by loretoparisi** [comment]
**[UPDATE]**
If I add both to `MessageGroups` target and `Pods/NMessenger` the flag to `NO`, I get some few errors, that Xcode helped me to resolve one by one in the editor.

**Comment by DavidSchechter** [comment]
Please look at this issue https://github.com/CocoaPods/CocoaPods/issues/5864

**Comment by loretoparisi** [comment]
@DavidSchechter yes thanks I have solved it, "faking" the requested Swift 3.0 update by Xcode, that didn't change anything actually, setting the `Use Legacy Swift Language Version` to `NO` explicitly, and solving by hand some errors in the `ViewController.swift`

This is the diff requested by Xcode8

```swift
$ git diff ViewController.swift 
diff --git a/examples/MessageGroups/MessageGroups/ViewController.swift b/examples/MessageGroups/MessageGroups/ViewController.swift
index 5674d41..d7fdd5e 100644
--- a/examples/MessageGroups/MessageGroups/ViewController.swift
+++ b/examples/MessageGroups/MessageGroups/ViewController.swift
@@ -24,14 +24,14 @@ class ViewController: NMessengerViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        self.senderSegmentedControl.frame = CGRect(x: self.segmentedControlPadding, y: UIApplication.sharedApplication().statusBarFrame.height, width: self.view.frame.width-2*self.segmentedControlPadding, height: self.segmentedControlHeight)
+        self.senderSegmentedControl.frame = CGRect(x: self.segmentedControlPadding, y: UIApplication.shared.statusBarFrame.height, width: self.view.frame.width-2*self.segmentedControlPadding, height: self.segmentedControlHeight)
         self.senderSegmentedControl.selectedSegmentIndex = 0
         self.navigationController?.view.addSubview(self.senderSegmentedControl)
         
         self.automaticallyAdjustsScrollViewInsets = false
     }
 
-    override func sendText(text: String, isIncomingMessage: Bool) -> GeneralMessengerCell {
+    override func sendText(_ text: String, isIncomingMessage: Bool) -> GeneralMessengerCell {
         
         //create a new text message
         let textContent = TextContentNode(textMessageString: text, currentViewController: self, bubbleConfiguration: self.sharedBubbleConfiguration)
@@ -41,9 +41,9 @@ class ViewController: NMessengerViewController {
         
         //add message to correct group
         if (self.senderSegmentedControl.selectedSegmentIndex == 0) { //incoming
-            self.postText(newMessage, isIncomingMessage: true)
+            self.postText(message: newMessage, isIncomingMessage: true)
         } else { //outgoing
-            self.postText(newMessage, isIncomingMessage: false)
+            self.postText(message: newMessage, isIncomingMessage: false)
         }
         
         return newMessage
@@ -66,7 +66,7 @@ class ViewController: NMessengerViewController {
             
             self.lastMessageGroup!.isIncomingMessage = isIncomingMessage
             self.messengerView.addMessageToMessageGroup(message, messageGroup: self.lastMessageGroup!, scrollsToLastMessage: false)
-            self.messengerView.addMessage(self.lastMessageGroup!, scrollsToMessage: true, withAnimation: isIncomingMessage ? .Left : .Right)
+            self.messengerView.addMessage(self.lastMessageGroup!, scrollsToMessage: true, withAnimation: isIncomingMessage ? .left : .right)
             
         } else {
             self.messengerView.addMessageToMessageGroup(message, messageGroup: self.lastMessageGroup!, scrollsToLastMessage: true)
@@ -90,7 +90,7 @@ class ViewController: NMessengerViewController {
      */
     private func createAvatar()->ASImageNode {
         let avatar = ASImageNode()
-        avatar.backgroundColor = UIColor.lightGrayColor()
+        avatar.backgroundColor = UIColor.lightGray
         avatar.preferredFrameSize = CGSize(width: 20, height: 20)
         avatar.layer.cornerRadius = 10
         return avatar
```
